src/client/CustomerClientApi.ts

```diff
@@ -0,0 +1,40 @@
+import { Customer, CustomersApi, SearchCustomersRequest, SearchCustomersResponse } from 'square-connect';
+import { IFindOrCreateCustomerRequest } from '../interface';
+
+export class CustomerClientApi extends CustomersApi {
+    async findOrCreateCustomer(findOrCreateCustomerRequest: IFindOrCreateCustomerRequest): Promise<Customer> {
+        const { phone_number, email_address }: IFindOrCreateCustomerRequest = findOrCreateCustomerRequest;
+
+        const searchCustomerReq: SearchCustomersRequest = {
+            query: {
+                filter: {
+                    phone_number: {
+                        exact: phone_number,
+                    },
+                },
+                sort: {
+                    field: 'CREATED_AT',
+                    order: 'DESC',
+                },
+            },
+        };
+
```

Comment:
just to make easier:

if (!email_address) {
return this.createCustomer(findOrCreateCustomerRequest)?
}

Reply:
No, because search customer may be undefined and  will need to  call create customer